tests/e2e_test.go

```diff
@@ -1004,3 +1006,85 @@ func TestConsistentSnapshotFlip(t *testing.T) {
 		t.Fatal("expected 2.snapshot.json in consistent snapshotted repo")
 	}
 }
+
+func TestSignWithEcdsaHexHSM(t *testing.T) {
+	ctx := context.Background()
+	td := t.TempDir()
+
+	rootCA, rootSigner, err := CreateRootCA()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	testTarget := filepath.Join(td, "foo.txt")
+	targetsConfig := map[string]json.RawMessage{testTarget: nil}
+
+	if err := os.WriteFile(testTarget, []byte("abc"), 0600); err != nil {
+		t.Fatal(err)
+	}
+
+	rootSerial, err := CreateTestHsmSigner(td, rootCA, rootSigner)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	snapshotKey := createTestSigner(t)
+	timestampKey := createTestSigner(t)
+
+	// Initialize succeeds with consistent snapshot off.
+	app.ConsistentSnapshot = false
+	if err := app.InitCmd(ctx, td, "", 1, targetsConfig, snapshotKey, timestampKey); err != nil {
+		t.Fatal(err)
+	}
+
+	// Sign root
+	rootKey, err := GetTestHsmSigner(ctx, td, *rootSerial)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := app.SignCmd(ctx, td, []string{"root"}, rootKey); err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that root has one signature using hex-encoded key.
```

Comment:
I'm missing how we know the key is hex-encoded. Is that because of the existing `keys. EcdsaTufKey` relies on `elliptic.Marshal` to generate the byte array, which is then hex-encoded at a later stage?
Would it be worth verifying that the key is indeed hex-encoded in `root.json`? I'm not fully sure where the real encoding to JSON happens, but I guess in `app.SignCmd`.

When the PEM encoding gets implemented, how will that look, will there be a different `GetTestHsmPemSigner`? 

Sorry if I'm asking stupid questions as I'm not fully up to speed yet with this code base.

Reply:
>  Is that because of the existing keys. EcdsaTufKey relies on elliptic.Marshal to generate the byte array, which is then hex-encoded at a later stage?

yep -- correct `EcdsaTufKey` generates hex-encoded keys. 

Yeah! I think you're right, I think I should have manually unmarshalled the key in the test and not pre-sumed the test will work because of the implementation. 

> When the PEM encoding gets implemented, how will that look, will there be a different GetTestHsmPemSigner?

My actual plan is that I will change `EcdsaTufKey`'s signature to take a `deprecated` parameter. See how it works here: https://github.com/sigstore/root-signing/pull/372/files. Then I'll add a `deprecated` parameter to to InitCmd that will flip the behavior.

That way, when we `Init` the repository from the current HSM keys, we will flip deprecated to `false` and generate them as new PEM keys. This will add only new PEM keys. The old hex-keys get [revoked](https://github.com/sigstore/root-signing/blob/f64d8fc4c71f0b4608e82d57aaae8e1f0142183a/cmd/tuf/app/init.go#L173) implicitly since they aren't added.

Gonna update the test for explicit hex-unmarshalling!

Reply:
https://github.com/sigstore/root-signing/pull/374/commits/998bfcb67edcab73c430145e8402fc792c908fcc

This change uses the explicit deprecated ECDSA verifier from go-tuf. This one expects the hex-encoded public key.
https://github.com/theupdateframework/go-tuf/blob/a9ddd8963324a7ba011317e8403bc889df367df0/pkg/keys/deprecated_ecdsa.go#L33

Reply:
Awesome, thanks for the clarifications.